### DISCUSSION TITLE
Google Auth return 401

### DISCUSSION THREAD

**Discussion by lrnzlo** [supabase/supabase#5826] [Questions]
```js
const supabaseUrl = 'https://hstmsonmjfkftbebbmph.supabase.co'
const supabaseAnonKey =
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhzdG1zb25tamZrZnRiZWJibXBoIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NDY0MDM2MDIsImV4cCI6MTk2MTk3OTYwMn0.KdMZcRLyFp8DLqR2IWbkI8HzzTuG96igFe8hJMrhl3EeyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhzdG1zb25tamZrZnRiZWJibXBoIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NDY0MDM2MDIsImV4cCI6MTk2MTk3OTYwMn0.KdMZcRLyFp8DLqR2IWbkI8HzzTuG96igFe8hJMrhl3E'
const client = createClient(supabaseUrl, supabaseAnonKey)

export async function signInWithGoogle() {
  const {user, session,  error } = await client.auth.signIn({
    provider: 'google',
  })
  console.log(user,session, error)
  
}
```
When I call this function to logged in with google it returns me 2 error message:
```bash
GET https://hst......supabase.co/auth/v1/user 401    helpers.ts:40
Error getting session from URL. {message: Invalid authentication credential status: 401}
```
But if I decode access_token 
```http://localhost:3000/#access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdW......```
on JWT.io it tells me that I'm authenticated as xxx@gmail.com and everything seems fine.

Someone that can help me?
This things is driving me insane

**Comment by lrnzlo** [comment]
I've try to replace the AnonKey with the ServiceKey and everything work fine. But that's a problem if I need to integrate this in a frontend framework like vue beacuse the key being exposed to the client side. Solution? In the docs it's written that the anon key is fine for this operation

**Comment by GaryAustin1** [comment]
Is some of your code running on server, or what environment are you running in?
Here are some issues with session error from url( but not sure they would even help depending on your environment):
https://github.com/supabase/gotrue-js/issues/23
https://github.com/supabase/supabase/issues/1693

**Reply by lrnzlo** [reply]
Everything is running as a vue 3 app
